Angular material 1.0.0

gpittarelli opened the pull request:
angular-material 1.0.0 reworked toasts by moving most of their styling
onto a wrapper div that they're (optionally) created in. However, this
wrapper div also has additional, reactive layout rules that don't make
sense for our site, so this change disables the wrapper div (autoWrap
setting in app.notifications.js) and re-adds to .toast the styles that
used to be on md-toast elements by default.

The toast background color and box-shadow are now always set by
_theme.scss (previously we were using the angular-material defaults).

mcapu commented:
Some things that need to be fixed (might be more, but I can't find them at the moment):

* `md-check` has now a different, dark check color for some reason (http://i.imgur.com/ndSGjcl.png instead of the old http://i.imgur.com/nmajPTB.png).
* `md-check`, as well as other components like radio buttons, are now bigger. This causes the slot button to look cramped (http://i.imgur.com/CXow6xO.png instead of the old http://i.imgur.com/7pAoRZn.png).
* Input fields (maybe other things as well) have now an element inserted after them, `.md-errors-spacer`, that causes this: http://i.imgur.com/lea7dOT.png, http://i.imgur.com/MeVJl0r.png.
* Toast action button feels too close to the right edge: http://i.imgur.com/t0E0bDo.png. I don't know whether this is intentional or not.

Apart from that, there are some style guidelines that aren't really written anywhere -which I should do soon:

* Use the `flex` mixin when dealing with flex display elements (https://github.com/gpittarelli/Frontend/blob/angular-material-1.0.0/src/scss/main/_elements.scss#L186). This will add some css rules that aren't 100% needed, but helps with readability.
* Unless there's a better reason for it in this case, extend `%md-whiteframe` when you want to apply shadows (extend `.sheet` if you want it to be a sheet, but that also means rounded corners and more stuff, which isn't good here).

gpittarelli commented:
> * `md-check` has now a different, dark check color for some reason (http://i.imgur.com/ndSGjcl.png instead of the old http://i.imgur.com/nmajPTB.png).

The checkmarks are now `$sheet-color`, as we agreed upon in slack

> * `md-check`, as well as other components like radio buttons, are now bigger. This causes the slot button to look cramped (http://i.imgur.com/CXow6xO.png instead of the old http://i.imgur.com/7pAoRZn.png).

Fixed for the checkboxes on slot buttons, left the same for radio/checkboxes on settings pages.

> * Input fields (maybe other things as well) have now an element inserted after them, `.md-errors-spacer`, that causes this: http://i.imgur.com/lea7dOT.png, http://i.imgur.com/MeVJl0r.png.

Reading the angular-material source, I believe `.md-errors-spacer` to be only created by `md-input`s. See angular/material#6214. Until that is fixed, I forced `max-height: 0` on those elements so that they will only take up space if they actually have some contents (such as the char count for the chat box).

Moving the char count has been a desired feature, so we got that for free :) Though there was extra spacing I fixed with a `margin-bottom: 0` (previously 24px) on the chat box's input field.

> * Toast action button feels too close to the right edge: http://i.imgur.com/t0E0bDo.png. I don't know whether this is intentional or not.

This was fixed before, I suspect you were looking at my test site which wasn't up to date.

> * Use the `flex` mixin when dealing with flex display elements (https://github.com/gpittarelli/Frontend/blob/angular-material-1.0.0/src/scss/main/_elements.scss#L186). This will add some css rules that aren't 100% needed, but helps with readability.
> * Unless there's a better reason for it in this case, extend `%md-whiteframe` when you want to apply shadows (extend `.sheet` if you want it to be a sheet, but that also means rounded corners and more stuff, which isn't good here).

Both of these have been addressed
